cpp/log/etcd_consistent_store-inl.h

```diff
@@ -648,32 +626,38 @@ util::Status EtcdConsistentStore<Logged>::CleanupOldEntries() {
   LOG(INFO) << "Cleaning old entries up to and including sequence number: "
             << clean_up_to_sequence_number;
 
-  std::vector<EntryHandle<Logged>> sequenced_entries;
-  util::Status status(GetSequencedEntries(&sequenced_entries));
+  EntryHandle<ct::SequenceMapping> sequence_mapping;
+  util::Status status(GetSequenceMapping(&sequence_mapping));
   if (!status.ok()) {
-    LOG(WARNING) << "Couldn't get sequenced entries: " << status;
+    LOG(WARNING) << "Couldn't get sequence mapping: " << status;
     return status;
   }
 
-  for (auto& entry : sequenced_entries) {
-    const uint64_t sequence_number(entry.Entry().sequence_number());
-
-    // return if we've hit the limit
-    if (entry.Entry().sequence_number() > clean_up_to_sequence_number) {
-      return util::Status::OK;
-    }
+  size_t i;
```

Comment:
I saw the use of ```i``` below, and I'm not sure I like it. :smiley: 

Either it's going to be one past ```clean_up_to_sequence_number```, or we'll have bailed out of this method altogether, right?